File: documentation-examples-java/src/java/dev/typr/foundations/docs/duckdb/IntegerTypesUnsigned.java
package dev.typr.foundations.docs.duckdb;

import dev.typr.foundations.DuckDbType;
import dev.typr.foundations.DuckDbTypes;
import dev.typr.foundations.data.Uint1;
import dev.typr.foundations.data.Uint4;

import java.math.BigInteger;

@SuppressWarnings("unused")
public class IntegerTypesUnsigned {
    //start
    DuckDbType<Uint1> utinyType = DuckDbTypes.utinyint;
    DuckDbType<Uint4> uintType = DuckDbTypes.uinteger;
    DuckDbType<BigInteger> uhugeType = DuckDbTypes.uhugeint;
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/duckdb/IntegerTypesUnsigned.scala
package dev.typr.foundations.docs.duckdb
import dev.typr.scalafoundations.*
import dev.typr.scalafoundations.data.*


import java.math.BigInteger

@SuppressWarnings(Array("unused"))
object IntegerTypesUnsigned:
  //start
  val utinyType: DuckDbType[Uint1] = DuckDbTypes.utinyint
  val uintType: DuckDbType[Uint4] = DuckDbTypes.uinteger
  val uhugeType: DuckDbType[BigInteger] = DuckDbTypes.uhugeint
  //stop
